src/shared/bootspec.c

```diff
@@ -841,6 +851,110 @@ static int find_uki_sections(
         return 0;
 }
 
+static int find_addon_sections(
+                int fd,
+                const char *path,
+                char **ret_cmdline) {
+
+        _cleanup_free_ IMAGE_SECTION_HEADER *sections = NULL;
+        _cleanup_free_ PeHeader *pe_header = NULL;
+        int r;
+
+        r = find_sections(fd, path, &sections, &pe_header);
+        if (r < 0)
+                return r;
+
+        r = find_cmdline_section(fd, path, sections, pe_header, ret_cmdline);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static int boot_entries_find_unified_addons(
+                BootConfig *config,
+                int d_fd,
+                const char *addon_dir,
+                const char *root,
+                BootEntryAddon **addons) {
```

Comment:
Oh, this is supposed to be a list. Please don't use linked lists for anything. Just return an array and a count.
The array should be resized using REALLOC_ARRAY when it is being read. This will make the code faster and simpler.